### PR TITLE
Feat/plug map energy layer to bdd

### PR DESCRIPTION
**Pull request by nkokla**

### CODE REVIEW

File: src/components/Map/components/LegendEntry.styled.ts [hunk 1]
```diff
@@ -11,8 +23,13 @@ export const LabelLegendWrapper = styled.div`
     align-items: baseline;
   }
 `;
-export const LabelLegend = styled.div`
+export const LabelLegend = styled.div<{ className?: string }>`
```
Comment (desoindx) [inline src/components/Map/components/LegendEntry.styled.ts:26]: Weird to have this classname attribute with a margin. And I think it's not used

File: src/components/Map/components/LegendEntry.styled.ts [hunk 2]
```diff
@@ -1,5 +1,17 @@
...
+export const LabelLegendInputLabelWrapper = styled.div`
+  margin-left: 4px;
+`;
+export const LabelLegendInputLabel = styled.div`
+  display: flex;
+`;
+export const LabelLegendDescription = styled.div`
+  margin-left: 8px;
+  font-size: 13px;
+  line-height: 1.5;
```
Comment (desoindx) [inline src/components/Map/components/LegendEntry.styled.ts:12]: 1.5 what ?

File: src/components/Map/components/LegendGroupLabel.tsx [hunk 3]
```diff
@@ -9,6 +9,7 @@ import {
 export type TypeGroupLegend = {
   id: string;
   subLegend?: string;
+  subLegendTxt?: string;
```
Comment (desoindx) [inline src/components/Map/components/LegendGroupLabel.tsx:12]: What's the difference between both ?

File: src/services/tiles.ts [hunk 4]
```diff
@@ -163,21 +230,55 @@ const tilesInfo: Record<
...
-    debug && console.info(`Indexing tiles for ${type} with ${table}...`);
-    const getter =
-      source === 'airtable'
-        ? getObjectIndexFromAirtable
-        : getObjectIndexFromDatabase;
-    getter(table, options, properties).then((result) => {
-      allTiles[type as DataType] = result;
-      debug && console.info(`Indexing tiles for ${type} with ${table} done`);
-    });
+    if (!allTiles[type as DataType]) {
```
Comment (desoindx) [inline src/services/tiles.ts:258]: I don't get this ?

File: src/services/tiles.ts [hunk 5]
```diff
@@ -163,21 +230,55 @@ const tilesInfo: Record<
...
+      );
+    } else {
+      debug &&
+        console.info(
+          `Indexing tiles for ${type} with ${table} already completed`
+        );
+    }
   }
 );
+Promise.all(promiseGetters).then(
```
Comment (desoindx) [inline src/services/tiles.ts:279]: Why ?

File: src/components/Map/components/LegendEntry.styled.ts (hunk 2 above)
Comment (nkokla) [reply src/components/Map/components/LegendEntry.styled.ts:12]: 1.5 of Font-size (It is an proportional value) 

From MDN > https://developer.mozilla.org/fr/docs/Web/CSS/line-height
`Unitless values: use this number multiplied by the element's font size`

File: src/components/Map/components/LegendGroupLabel.tsx (hunk 3 above)
Comment (nkokla) [reply src/components/Map/components/LegendGroupLabel.tsx:12]: - subLegend is a ref to an 'React Functional Component' (For advanced render, like the Box under 'Bâtiments chauffés au gaz collectif' or 'Bâtiments chauffées au fioul collectif')
- subLegendTxt is just a string

File: src/services/tiles.ts (hunk 4 above)
Comment (nkokla) [reply src/services/tiles.ts:258]: Ho, yes... It an old test. I clear... 👍

File: src/services/tiles.ts (hunk 5 above)
Comment (nkokla) [reply src/services/tiles.ts:279]: It's just for a debug log when all indexing is done. 
I find it useful, but if you prefer we can delete it.

Comment (desoindx) [reply src/services/tiles.ts:279]: okay for me

**Review by desoindx — APPROVED** [review]

**Review by desoindx — APPROVED** [review]

**Pull request merged**